src/main/java/net/sf/jabref/external/ExternalFileTypes.java

```diff
@@ -184,7 +170,7 @@ public ExternalFileType getExternalFileTypeForName(String filename) {
                 foundType = type;
```

Comment:
No. First, I missed the point of your comment, got confused by the shaded text stating that ExternalFileType was returned. Here there's a mechanism (unknown why), which honours the longest extension string. Therefore, all types must be checked.